core/blockchain.go

```diff
@@ -1145,9 +1142,24 @@ func (self *BlockChain) reorg(oldBlock, newBlock *types.Block) error {
 		}
 	}
 
-	if glog.V(logger.Debug) {
+	oldLen := len(oldChain)
+
+	if glog.V(logger.Debug) || oldLen > 63 {
 		commonHash := commonBlock.Hash()
-		glog.Infof("Chain split detected @ %x. Reorganising chain from #%v %x to %x", commonHash[:4], numSplit, oldStart.Hash().Bytes()[:4], newStart.Hash().Bytes()[:4])
+		commonNumber := commonBlock.Number()
```

Comment:
Since both `commonHash` and `commonNumber` is used only once below, I'd just inline these there and not declare separate variables. You can do `commonBlock.Hash().Bytes()[:4]` to get the first 4 bytes inline.